receiver/frequency_far_jump/frequency_far_jump.js

```diff
@@ -0,0 +1,55 @@
+/*
+ * Plugin: Jump to a frequency outside the boundaries of the selected profile by typing it in the receiver's frequency dial.
+ * Requires the option 'Allow users to change center frequency' to be enabled in the admin panel.
+ *
+ * License: Apache License 2.0
+ * Copyright (c) 2024 Dimitar Milkov, LZ2DMV
+ */
+
+Plugins.frequency_far_jump.no_css = true;
+
+Plugins.frequency_far_jump.init = async function () {
+
+  if (!Plugins.isLoaded('utils', 0.1)) {
+
+    await Plugins.load('https://0xaf.github.io/openwebrxplus-plugins/receiver/utils/utils.js');
+
+    if (!Plugins.isLoaded('utils', 0.1)) {
+      console.error('Plugin "frequency_far_jump" depends on "utils >= 0.1".');
+      return false;
+    } else {
+      Plugins._debug('Plugin "utils" has been loaded as dependency.');
+    }
+  }
+
+  Plugins.utils.wrap_func(
+    'set_offset_frequency',
+    function (orig, thisArg, args) {
+      var to_what = Math.round(args[0]);
+
+      if (typeof(to_what) == 'undefined') return;
+
+      // The frequency is outside the boundaries of the current profile
+      if (to_what > bandwidth / 2 || to_what < -bandwidth / 2) {
+        // to_what is an offset, so we need to add the current full frequency (center_freq) to it
+        var f = center_freq + to_what;
+        var k = $('#openwebrx-panel-receiver').demodulatorPanel().getMagicKey();
```

Comment:
Would be nice to have a "notification" or at least console.error() or something in case the magic key is not provided. Otherwise the admins and/or users will be confused and a lot of questions will come.
(if you are going to implement the notifications, see other plugins or contact me)

Reply:
Yes, but the instance might not have a magic key configured, and in thst case the user won't be required to supply one, i.e. a hypothetical check for empty value will come true, but the user's intention was to use the instance without a key, because the frequency change option of the instance is unlocked (no magic key in the admin panel).

We hsve an event being emitted for when the frequency changes, but this event is emitted for any frequency change, including when we are switching frequency within the limits of the current profile, and not only when we switch to a frequency outside the profile using the WS interface this plugin utilizes.